Marlin/src/lcd/ultralcd.cpp

```diff
@@ -5004,7 +5004,7 @@ void lcd_init() {
 
   lcd_implementation_init();
 
-  #if ENABLED(NEWPANEL)
+  #if ENABLED(NEWPANEL) || ENABLED(U8GLIB_SH1106_EINSTART)
```

Comment:
It's more correct to have `U8GLIB_SH1106_EINSTART` activate `NEWPANEL` in `Conditionals_LCD.h` as with the other `SH1106` options. The `NEWPANEL` conditional indicates a panel that uses digital buttons rather than a shift register.

Reply:
I wasn't sure, but this *did* get LCD functional until I moved to native fastio, now it's gone.

I'll try switching the invocation to Conditiontals_LCD.h and invoke NEWPANEL and see if that fixes things up.

Would it be more or less hassle to just add a conditional to one of the existing SH1106 u8g constructors? After looking at the other implementations they're the same but just don't have RES and DC defined.

Might be more hassle than it's worth, I'm not sure if there's bigger plans for the LCD code and I don't want to make a mess of things.

Reply:
I moved it up with the rest, now I just need to establish what is the correct combo of LCD options to invoke to make it work. 

From what I gather, the LCD defines in Configuration.h are picked up by Conditionals_LCD.h to enable the various DOGLCD, ULTIPANEL, NEWPANEL options, which are then compiled in via ultralcd.cpp and ultralcd_impl_DOGM.h.

In my initial brute force attack I got it working by adding a selectors at:

 - Conditionals_LCD.h:166 (enabling ULTRA_LCD and DOGLCD)
 - Conditionals_LCD.h:194 (enabling ULTIPANEL)
 - ultralcd.cpp:4997 (set SET_INPUT(BTN_##)...)
 - ultralcd_impl_DOGM.h:184 (actual u8g constructor)

Now I've got a single selector activating all the options in Conditionals_LCD:208 (ULTRA_LCD, DOGLCD, ULTIPANEL, NEWPANEL), no changes in ultralcd.cpp, and the constructor in place in ultralcd_impl_DOGM.h. No LCD.

I removed the if from ultralcd.cpp:4997 since by enabling NEWPANEL in Conditionals_LCD.h I should get the same code path, and I'm getting a large enough compile that I'm confident the includes are happening, but something isn't happy.